onnx/test/checker_test.py

```diff
@@ -217,6 +217,39 @@ def test_nested_graph(self):  # type: () -> None
         checker.check_graph(graph)
         #self.assertRaises(checker.ValidationError, checker.check_graph, graph)
 
+    def test_nested_graph_without_subgraph_input_shape(self):  # type: () -> None
+        n1 = helper.make_node(
+            "Scale", ["X"], ["Y"], scale=2., name="n1")
+        n2 = helper.make_node(
+            "Scale", ["Y"], ["Z"], scale=3., name="n2")
+
+        input_x = onnx.ValueInfoProto()
+        input_x.name = "X"
+        graph = helper.make_graph(
+            [n1, n2],
+            "nested",
+            inputs=[
+                input_x
+            ],
+            outputs=[
+                helper.make_tensor_value_info("Z", TensorProto.FLOAT, [1, 2])
+            ]
+        )
+
+        i1 = helper.make_node(
+            "If", ["cond"], ["Z"], then_branch=graph, else_branch=graph)
+
+        graph = helper.make_graph(
+            [i1],
+            "test",
+            inputs=[
+                helper.make_tensor_value_info("cond", TensorProto.BOOL, [1])
```

Comment:
Define X?

Reply:
Added, also added to the original nested_graph test case that I copied from.